styx-common/src/main/java/com/spotify/styx/state/RunState.java

```diff
@@ -182,13 +183,14 @@ public RunState info(WorkflowInstance workflowInstance, Message message) {
     }
 
     @Override
-    public RunState dequeue(WorkflowInstance workflowInstance) {
+    public RunState dequeue(WorkflowInstance workflowInstance, Set<String> resourceIds) {
       switch (state()) {
         case QUEUED:
           return state(
               PREPARE,
               data().builder()
                   .retryDelayMillis(empty())
+                  .resourceIds(resourceIds)
```

Comment:
`resourceIds` can be null here, but I believe the builder translates it into a Optional.empty. I'd still prefer to explicitly receive an Optional.

Reply:
Indeed translated by builder same way as in `submit`; will leave it consistent with that for this PR.